File: python/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        '''
        Time: O(n + n) -> split is n time and then we go to each word and check if its not empty which is also n
            : O(n)
        Space: O(n) create space for length of string to split then create new array to store teh words without the empty spaces 
        '''
        words = [word for word in s.split(" ") if word]
        return len(words[-1])

File: javascript/0058-length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
    // Time: O(n)
    // Space: O(1)

    const split_s = s.split(" ");
    let last_word_len = 0;
    for (let i = 0; i < split_s.length; i++) {
        if (split_s[i].length > 0) {
            last_word_len = split_s[i].length;
        }
    }

    return last_word_len;
};
